### PR TITLE
ARROW-8777: [Rust] Parquet.rs does not support reading fixed-size binary fields.

### PR DESCRIPTION
**Pull request by maxburke**
Fixed length binary fields require some extra context to be passed to
the data conversion functions; the byte width of the data field.
    
This change also adds more specific converters for reading timestamp
types into Arrow. Previously an error would be raised because the schema
type (ie: Timestamp(Microseconds, _)) was different than the
converter-generated column type (UInt64).

### CODE REVIEW

**Comment by maxburke** [comment]
Apologies for the hiccups in getting this green but it's sorted now.

File: rust/parquet/src/arrow/array_reader.rs [hunk 1]
```diff
@@ -1042,16 +1079,22 @@ mod tests {
             ",
                 $physical_type, $logical_type_str
             );
+            eprintln!("1");
```
Comment (nevi-me) [inline rust/parquet/src/arrow/array_reader.rs:1082]: I think you committed these by mistake?

**Review by nevi-me — COMMENTED** [review]
I'm happy with the changes, but please remove the `println`s, I think you added them by mistake

File: rust/parquet/src/arrow/array_reader.rs (hunk 1 above)
Comment (maxburke) [reply rust/parquet/src/arrow/array_reader.rs]: Yup, my mistake. Removed.

**Review by nevi-me — APPROVED** [review]
LGTM

**Pull request closed without merging**